src/resolution_functions/models/polynomial.py

```diff
@@ -14,24 +21,138 @@
 
 @dataclass(init=True, repr=True, frozen=True, slots=True, kw_only=True)
 class PolynomialModelData(ModelData):
+    """
+    Data for the `PolynomialModel1D` model.
```

Comment:
Maybe we should be using ``:class:`PolynomialModel1D` `` in places like this. It makes the source less pleasant to read but the autodocs nicer... a bit of a trade-off.

Reply:
If you mean cross-referencing, `numpydoc` handles that automagically just by using single backticks.